### PR TITLE
[api] fix issue #3215

### LINKED ISSUE #3215: eth_synching not implemented
Please add eth_syncing endpoint support to native web3 API server.

Currently this endpoint always returns "false" `{"jsonrpc":"2.0","id":1,"result":false}` since in the code the endpoint is hardcoded to do so `func (svr *Web3Server) isSyncing() (interface{}, error) {
	return false, nil
}
`

Can you please implement this endpoint soon? Node operators rely heavily on this endpoint to check if the node is running correctly when syncing from height 0

### PR DESCRIPTION
**Pull request by Liuhaai**
# Description

fix #3215 

 - startingBlock:       block number this node started to synchronise from
 - currentBlock:        block number this node is currently importing
- highestBlock:        block number of the highest block header this node has received from peers

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/coreservice.go [hunk 1]
```diff
@@ -134,6 +134,8 @@ type (
 		ReadContractStorage(ctx context.Context, addr address.Address, key []byte) ([]byte, error)
 		// SimulateExecution simulates execution
 		SimulateExecution(context.Context, address.Address, *action.Execution) ([]byte, *action.Receipt, error)
+		// SyningProgress returns the syncing status of node
+		SyningProgress() (uint64, uint64, uint64)
```
Comment (dustinxie) [inline api/coreservice.go:138]: `SyncingStatus`

File: blockchain/blockdao/blockdao.go [hunk 2]
```diff
@@ -48,6 +48,7 @@ type (
 		GetActionByActionHash(hash.Hash256, uint64) (action.SealedEnvelope, uint32, error)
 		GetReceiptByActionHash(hash.Hash256, uint64) (*action.Receipt, error)
 		DeleteBlockToTarget(uint64) error
+		StartingHeight() uint64
```
Comment (dustinxie) [inline blockchain/blockdao/blockdao.go:51]: `StartingHeight` feels more like a concept belong to the blockchain, so prefer to have this in `Blockchain` module, simply check `blockdao`'s height at the end of Start:
```
func (bc *blockchain) Start() {
// current code
bc.startHeight, err = bc.dao.Height()
}
```

File: blockchain/blockdao/blockdao.go [hunk 3]
```diff
@@ -60,14 +61,15 @@ type (
...
-		tipHeight    uint64
+		blockStore     filedao.FileDAO
+		indexers       []BlockIndexer
+		timerFactory   *prometheustimer.TimerFactory
+		lifecycle      lifecycle.Lifecycle
+		headerCache    *cache.ThreadSafeLruCache
+		bodyCache      *cache.ThreadSafeLruCache
+		footerCache    *cache.ThreadSafeLruCache
+		tipHeight      uint64
+		startingHeight uint64
```
Comment (dustinxie) [inline blockchain/blockdao/blockdao.go:72]: move to blockchain

File: api/coreservice.go [hunk 4]
```diff
@@ -1635,3 +1637,10 @@ func (core *coreService) SimulateExecution(ctx context.Context, addr address.Add
 	exec.SetGasLimit(core.bc.Genesis().BlockGasLimit)
 	return core.sf.SimulateExecution(ctx, addr, exec, core.dao.GetBlockHash)
 }
+
+// SyningProgress returns the syncing status of node
+func (core *coreService) SyningProgress() (uint64, uint64, uint64) {
+	return core.dao.StartingHeight(),
+		core.bc.TipHeight(),
+		core.bs.TargetHeight()
```
Comment (dustinxie) [inline api/coreservice.go:1645]: should return `"result": false` if the chain is not syncing (reaches highest block)

File: api/web3server.go [hunk 5]
```diff
@@ -565,7 +571,12 @@ func (svr *Web3Server) getProtocolVersion() (interface{}, error) {
 }
 
 func (svr *Web3Server) isSyncing() (interface{}, error) {
-	return false, nil
+	start, curr, highest := svr.coreService.SyningProgress()
+	return &syncingObject{
+		startingBlock: uint64ToHex(start),
+		currentBlock:  uint64ToHex(curr),
+		highestBlock:  uint64ToHex(highest),
+	}, nil
```
Comment (dustinxie) [inline api/web3server.go:579]: should return `"result": false` if the chain is not syncing any more (reaches highest block)

File: api/coreservice.go (hunk 1 above)
Comment (dustinxie) [reply api/coreservice.go:138]: `Syncing` not `Syning`

**Review by dustinxie — APPROVED** [review]

File: api/web3server.go [hunk 6]
```diff
@@ -565,7 +571,15 @@ func (svr *Web3Server) getProtocolVersion() (interface{}, error) {
 }
 
 func (svr *Web3Server) isSyncing() (interface{}, error) {
-	return false, nil
+	start, curr, highest := svr.coreService.SyncingProgress()
+	if curr >= highest {
+		return false, nil
+	}
+	return &syncingObject{
+		startingBlock: uint64ToHex(start),
```
Comment (CoderZhi) [inline api/web3server.go:579]: what's the definition of `startingBlock`?

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**